minecraft/protocol/packet/compression.go

```diff
@@ -106,20 +108,43 @@ func (flateCompression) Compress(decompressed []byte) ([]byte, error) {
 
 // Decompress ...
 func (flateCompression) Decompress(compressed []byte, limit int) ([]byte, error) {
-	buf := bytes.NewReader(compressed)
-	c := flateDecompressPool.Get().(io.ReadCloser)
-	defer flateDecompressPool.Put(c)
+	r := flateDecompressPool.Get().(io.ReadCloser)
+	defer func() {
+		_ = r.Close()
```

Comment:
I am not sure if we need to Close() at all. For zlib, the Close before reading was necessary for the checksum, but I believe deflate has no such checksum. Given we don't need it, maybe Close() isn't necessary at all?